lesson26/src/js/login.js

```diff
@@ -0,0 +1,38 @@
+import * as validation from "./validation";
+
+const submitButton  = document.querySelector('[data-id="submit_button"]');
+
+submitButton.addEventListener("click", (e) => {
+    e.preventDefault();
+});
+
+const changeDisabledStatusSubmitButton = () => submitButton.disabled = validation.isInvalid();
+
+const inputSelector = document.querySelectorAll('input');
+
+for (const input of inputSelector){
+    input.classList.add("field-invalid");
+
+    input.addEventListener("blur", () => {
+        if(input.hasAttribute("required") && input.value.trim() === ""){
+            validation.showErrorMessage(input,"入力してください");
+        }else{
+            validation.removeErrorMessage(input);
+
+            input.type === "password" && validation.checkPassword(input);
+        }
+        changeDisabledStatusSubmitButton();
+    });
+
+    input.addEventListener("input" , () => {
+        if(input.nextElementSibling && input.nextElementSibling.classList.contains("field-invalid")){
+            input.type === "password" && validation.checkPassword(input);
+        }
```

Comment:
Q. Is this still okay?
```suggestion
    input.addEventListener("input" , () => {
        if(input.nextElementSibling.classList.contains("field-invalid")){
            input.type === "password" && validation.checkPassword(input);
        }
```

Reply:
Thanks for your comment! 

If `input.nextElementSibling` does not exist (no error indication), just `input.nextElementSibling.classList.contains("field-invalid")` will result in an error, so I have added a ` input.nextElementSibling` is also checked for the existence of `input.nextElementSibling`.

Reply:
Thank you for telling us about it :)
There was an omission in checking the operation🙇‍♀️ Sorry..🙏